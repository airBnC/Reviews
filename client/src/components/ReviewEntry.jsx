import React from 'react';
import PropTypes from 'prop-types';

import {
  ReviewEntryWrapper,
  ReviewUserWrapper,
  ReviewDateWrapper,
  ReviewTextWrapper,
} from './styles/ReviewEntry.styles';

const ReviewEntry = props => (
  <ReviewEntryWrapper>
    <ReviewUserWrapper> {props.review.first_name} </ReviewUserWrapper>
    <ReviewDateWrapper> {props.review.review_date} </ReviewDateWrapper>
    <ReviewTextWrapper> {props.review.review_text} </ReviewTextWrapper>
  </ReviewEntryWrapper>
);

ReviewEntry.propTypes = {
  review: PropTypes.shape({
    review_date: PropTypes.string,
    review_text: PropTypes.string,
    first_name: PropTypes.string,
    last_name: PropTypes.string,
  }),
};

ReviewEntry.defaultProps = {
  review: {
    review_date: '2016-11-04',
    review_text: 'Laborum eius id rerum et in ratione',
    first_name: 'Toby',
    last_name: 'Keith',
  },
};

export default ReviewEntry;
