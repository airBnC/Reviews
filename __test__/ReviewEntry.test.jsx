import React from 'react';
import { configure, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import ReviewEntry from '../client/src/components/ReviewEntry';

configure({ adapter: new Adapter() });

const dummyReview = {
  review_date: '2017-02-12T08:00:00.000Z',
  review_text: 'Debitis sequi id nam optio iste molestias dolores dolorem ex.',
  first_name: 'Peggie',
  last_name: 'Oberbrunner',
};

const dummyKey = 3;

describe('ReviewEntry', () => {
  it('should render correctly', () => {
    const output = shallow(<ReviewEntry key={dummyKey} review={dummyReview} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
