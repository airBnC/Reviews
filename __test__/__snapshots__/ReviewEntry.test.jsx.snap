// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewEntry should render correctly 1`] = `
<styled.div>
  <styled.div>
     
    Peggie
     
  </styled.div>
  <styled.div>
     
    2017-02-12T08:00:00.000Z
     
  </styled.div>
  <styled.div>
     
    Debitis sequi id nam optio iste molestias dolores dolorem ex.
     
  </styled.div>
</styled.div>
`;
